### PR TITLE
Minidlna

### PR DESCRIPTION
**Pull request by julien-lecomte**
Allows user to install and setup a working MiniDLNA server. Registers ports in /etc/services, and setups the ferm firewall.

### CODE REVIEW

File: ansible/playbooks/srv/minidlna.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (drybjed) [inline ansible/playbooks/srv/minidlna.yml]: Looks like this is a real file? It should be a symlink to `../service/minidlna.yml` instead, so that the two playbooks don't get desynchronized in the future.

File: ansible/roles/sysctl/templates/etc/sysctl.d/parameters.conf.j2 [hunk 2]
```diff
@@ -20,6 +20,7 @@
 {{ value -}}
 {%   endif %}
 {% endmacro %}
+
```
Comment (drybjed) [inline ansible/roles/sysctl/templates/etc/sysctl.d/parameters.conf.j2]: Why this change is here?

File: ansible/roles/minidlna/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+# List of media directories to scan and present to end user.
+#
+# If you want to restrict a media_dir to a specific content type, you can
+# prepend the directory name with a letter representing the type (A, P or V),
+# followed by a comma, as so:
+#
+# * "A" for audio    (eg. media_dir=A,/var/lib/minidlna/music)
+# * "P" for pictures (eg. media_dir=P,/var/lib/minidlna/pictures)
+# * "V" for video    (eg. media_dir=V,/var/lib/minidlna/videos)
+# * "PV" for pictures and video (eg. media_dir=PV,/var/lib/minidlna/digital_camera)
```
Comment (drybjed) [inline ansible/roles/minidlna/defaults/main.yml]: The examples here should reflect what needs to be put in the variable, so don't add `media_dir=` since this will be prefixed in the template automatically.

File: ansible/roles/minidlna/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+                                                                   # ]]]
+# .. envvar:: minidlna__allow [[[
+#
+# List of IP addresses or CIDR subnets which are allowed to connect to the
+# MiniDLNA instances over the network, on all hosts in the Ansible
+# inventory. This variable configures the firewall for all instances at the
+# same time, for individual instance configuration you should modify the
+# :envvar:`minidlna__ferm__dependent_rules` variable directly.
+minidlna__allow: {}
```
Comment (drybjed) [inline ansible/roles/minidlna/defaults/main.yml]: This should be a list (`[]`) and not a YAML dictionary (`[]`), otherwise the `ferm` role will not parse this correctly.

File: ansible/roles/minidlna/templates/etc/ansible/facts.d/minidlna.fact.j2 [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{# Copyright (C) 2021 Julien Lecomte <julien@lecomte.at>
```
Comment (drybjed) [inline ansible/roles/minidlna/templates/etc/ansible/facts.d/minidlna.fact.j2]: It would be better to make the fact a Python script. That way users can use `run-parts /etc/ansible/facts.d` to get a nice overview, or other services can benefit from this. The `postwhite` role has a nice example of a simple Python fact script.

The fact scripts should if possible gather information from the host itself. You could check the minidlna configuration file and find out the actual http_port in use.

Comment (julien-lecomte) [reply ansible/roles/minidlna/templates/etc/ansible/facts.d/minidlna.fact.j2]: I'll look into that.

**Comment by julien-lecomte** [comment]
I'd like to bump this MR so I can move onto the next ones. Thanks

**Comment by drybjed** [comment]
@julien-lecomte I started looking into the PR and I'll probably make a few changes, for example add the whole contents of the configuration file by default. Expect this to be merged in a few days tops.
